submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/ZipUtil.kt

```diff
@@ -39,4 +46,20 @@ object ZipUtil {
         zipEntry.time = 0
         return zipEntry
     }
+
+    internal fun checkZipIntegrity(
+        sourceDir: File,
+        zipFile: File,
+    ) {
+        ZipFile(zipFile).use { zip ->
+            listAllFilesWithRelPaths(sourceDir).forEach {
+                val (path, sourceFile) = it
+                val entry = zip.getEntry(path)
+
+                if (entry == null || sourceFile.md5() != zip.md5(entry)) throw CorruptedFileException(sourceFile)
+            }
+        }
+    }
+
+    private fun ZipFile.md5(entry: ZipEntry): String = getInputStream(entry).use { DigestUtils.md5Hex(it).uppercase() }
```

Comment:
with this change we are calculating the md5 twice, this is impacting significantly the zip time for large folder. It looks like a bad idea IMO, still if that is what need it well we can do it but IMO this is doing more harm than good to the system. Ugis will have lass optinion